### PR TITLE
nixos/systemd: document what `scriptArgs` is for

### PR DESCRIPTION
**Pull request by ncfavier**
Related to https://github.com/NixOS/nixpkgs/issues/154123, I guess

### CODE REVIEW

**Comment by ncfavier** [comment]
Would be nice to have manpage links working in `options.html`. I think we could

- split out the manpage → URL mapping into a JSON file or whatever lua can parse
- duplicate the `link-unix-man-references.lua` logic [here]( https://github.com/NixOS/nixpkgs/blob/8de585e60b7349fa6d02c2582dd07c0acc512d04/nixos/lib/make-options-doc/mergeJSON.py#L121)

**Review by pennae — COMMENTED** [review]
looks good. as for linking in `options.html`, we could move the manpage reference table to a special doc attribute that generates an MD footer with link targets? that way the renderers could remain agnostic and html output can get automatic linking with no extra effort.

**Review by pennae — APPROVED** [review]

File: doc/build-aux/pandoc-filters/link-unix-man-references.lua [hunk 1]
```diff
@@ -3,10 +3,31 @@ Turns a manpage reference into a link, when a mapping is defined below.
...
+  ["systemd.service(5)"] = "https://www.freedesktop.org/software/systemd/man/systemd.service.html",
+  ["systemd.slice(5)"] = "https://www.freedesktop.org/software/systemd/man/systemd.slice.html",
+  ["systemd.socket(5)"] = "https://www.freedesktop.org/software/systemd/man/systemd.socket.html",
   ["systemd.timer(5)"] = "https://www.freedesktop.org/software/systemd/man/systemd.timer.html",
+  ["systemd.unit(5)"] = "https://www.freedesktop.org/software/systemd/man/systemd.unit.html",
+  ["timesyncd.conf(5)"] = "https://www.freedesktop.org/software/systemd/man/timesyncd.conf.html",
+  ["tmpfiles.d(5)"] = "https://www.freedesktop.org/software/systemd/man/tmpfiles.d.html",
+  ["systemd.time(7)"] = "https://www.freedesktop.org/software/systemd/man/systemd.time.html",
+  ["systemd-fstab-generator(8)"] = "https://www.freedesktop.org/software/systemd/man/systemd-fstab-generator.html",
+  ["systemd-networkd-wait-online.service(8)"] = "https://www.freedesktop.org/software/systemd/man/systemd-networkd-wait-online.service.html",
```
Comment (roberth) [inline doc/build-aux/pandoc-filters/link-unix-man-references.lua:30]: Neat.

**Review by roberth — APPROVED** [review]

File: doc/build-aux/pandoc-filters/link-unix-man-references.lua (hunk 1 above)
Comment (roberth) [reply doc/build-aux/pandoc-filters/link-unix-man-references.lua:30]: Actually it doesn't seem to work? When I render the manual and load the options page, I don't see any replacements. Maybe this operates on the docbook, which is structured?
```python
    def manpage(self, page, section):
        title = f"<refentrytitle>{escape(page)}</refentrytitle>"
        vol = f"<manvolnum>{escape(section)}</manvolnum>"
        return f"<citerefentry>{title}{vol}</citerefentry>"
```

**Review by roberth — CHANGES_REQUESTED** [review]
May need a fix

File: doc/build-aux/pandoc-filters/link-unix-man-references.lua (hunk 1 above)
Comment (ncfavier) [reply doc/build-aux/pandoc-filters/link-unix-man-references.lua:30]: Yep, see https://github.com/NixOS/nixpkgs/pull/207268#issuecomment-1362940973

Comment (roberth) [reply doc/build-aux/pandoc-filters/link-unix-man-references.lua:30]: Ah I see. I thought you implemented that. I should've clicked the link in that comment.

**Review by roberth — APPROVED** [review]

**Comment by ncfavier** [comment]
@pennae that would require changing the manpage syntax to something like `[systemd.unit(5)]`, unless i'm misunderstanding? But then other documents would have to append the footer everywhere, which seems burdensome, unless there's an easy way to do that with pandoc?

**Comment by ncfavier** [comment]
> not necessarily

Well, if we want this to work with a footer (I'm assuming you're talking about https://pandoc.org/MANUAL.html#reference-links), the manpage references must have that shape, so we'd need a step that converts `` {manpage}`foo` `` to `[foo]` *in Markdown*, before we append the footer and render to HTML. But at that point it seems easier to just do the conversion from `` {manpage}`foo` `` to `<a href=...>foo</a>` directly, as a filter in the Markdown → HTML pipeline.

**Comment by ncfavier** [comment]
> Would be nice to have manpage links working in `options.html`.

Done in https://github.com/NixOS/nixpkgs/pull/208762

**Pull request merged**